kamon-core/src/main/java/kamon/jsr166/Striped64.java

```diff
@@ -182,6 +186,13 @@ final boolean casBase(long cmp, long val) {
     }
 
     /**
+     * CASes the base field.
```

Comment:
This code are using Java 8 Unsafe intrinsics (in this case lock xchg) and isn't supported in prior java versions.
Maybe we could check if support getAndSetObject in this way Unsafe.class.getMethod("getAndSetObject", Object.class, Long.TYPE,Object.class) and use this method, otherwise fallback to something like:

```java
 final long getAndSetBase(long val) {
  long v;
     do {
          v = UNSAFE.getLongVolatile(this, baseOffset);
      } while (!UNSAFE.compareAndSwapLong(this, baseOffset, v, val));
  return v;
 }
```
or simply use the code above (CAS version, I do not know if the code compiles ;)) and the same for the `final long getAndSet(long val)`